wagtail/snippets/blocks.py

```diff
@@ -17,7 +23,9 @@ def target_model(self):
     def widget(self):
         from wagtail.snippets.widgets import AdminSnippetChooser
 
-        return AdminSnippetChooser(self.target_model)
+        # Override the default icon with the icon for the target model
+        self.set_meta_options({"icon": self.target_model.snippet_viewset.icon})
```

Comment:
We can also just do `self.meta.icon = ...`, but seeing that we already have the method, I figured why not use it. It does require the `MUTABLE_META_ATTRIBUTES` override, though.